frontend/src/metabase/query_builder/components/view/ViewFooter/LeftViewFooterButtonGroup.tsx

```diff
@@ -1,58 +1,136 @@
+  const isVisualized = useSelector(getIsVisualized);
+
+  const data = useMemo(
+    () =>
+      [
+        isNative
+          ? null
+          : {
+              value: "editor",
+              label: (
+                <Tooltip label={t`Editor`}>
+                  <Icon
+                    name="notebook"
+                    onClick={() => {
+                      dispatch(setQueryBuilderMode("notebook"));
+                    }}
+                  />
+                </Tooltip>
+              ),
+            },
+        {
+          value: "table",
+          label: (
+            <Tooltip label={t`Results`}>
+              <Icon
+                aria-label={t`Switch to data`}
+                name="table2"
+                onClick={() => {
+                  dispatch(setUIControls({ isShowingRawTable: true }));
+                }}
+              />
+            </Tooltip>
+          ),
+        },
+        {
+          value: "visualization",
+          // TODO: also we need to add a spinner :boom:
+          label: (
+            <Tooltip label={t`Visualization`}>
+              <Icon
+                aria-label={t`Switch to visualization`}
+                name={vizIcon}
+                onClick={() => {
+                  dispatch(setUIControls({ isShowingRawTable: false }));
+                }}
+              />
+            </Tooltip>
+          ),
+        },
+      ].filter(isNotNull),
+    [dispatch, isNative, vizIcon],
+  );
 
   return (
-    <Group className={CS.flex1}>
-      <FooterButtonGroup>
-        <ViewButton
-          medium
-          labelBreakpoint="sm"
-          data-testid="viz-type-button"
-          active={isShowingChartTypeSidebar}
-          onClick={
-            isShowingChartTypeSidebar
-              ? () => dispatch(onCloseChartType())
-              : () => dispatch(onOpenChartType())
-          }
-        >
-          {t`Visualization type`}
-        </ViewButton>
-        <ViewButton
-          active={isShowingChartSettingsSidebar}
-          icon="gear"
-          iconSize={16}
-          medium
-          iconWithText
-          labelBreakpoint="sm"
+    <Flex gap="0.75rem">
+      {isVisualized && (
+        <EditorViewControl
+          value={isShowingRawTable ? "table" : "visualization"}
+          data={data}
+        />
+      )}
+      {(!isShowingRawTable || isVisualized) && (
+        <Button
+          radius="xl"
+          variant={isShowingChartSettingsSidebar ? "filled" : "default"}
+          styles={{
+            ...(!isShowingChartSettingsSidebar && {
+              root: {
+                backgroundColor: "var(--mb-color-brand-lighter)",
+                color: "var(--mb-color-brand)",
+                border: 0,
+
+                "&:hover": {
+                  backgroundColor: "var(--mb-color-focus)",
+                },
+              },
+            }),
+          }}
+          /* mah is a hack for 32px height button, we don't have it atm */
+          mah="xl"
```

Comment:
"Visualize" button uses a different way to achieve 32px. Let's make it consistent.

Reply:
done